File: python/clock/clock.py
class Clock(object):

    def minutesPerDay(self):
        return 24 * 60

    def __init__(self, hours, minutes):
        self.offset = (minutes + 60 * hours) % self.minutesPerDay()

    def __eq__(self, other):
        return self.offset == other.offset

    def __str__(self):
        hours = self.offset / 60
        minutes = self.offset % 60
        return "{:02d}:{:02d}".format(hours, minutes)

    def add(self, minutes):
        self.offset = (self.offset + minutes) % self.minutesPerDay()
        return self

File: csharp/clock/Clock.cs
using System;

public class Clock : IEquatable<Clock>
{
    private const int MinutesPerDay = 24 * 60;
    private const int MinutesPerHour = 60;
    private readonly int _minutes;

    public Clock(int hours) => _minutes = hours * MinutesPerHour % (MinutesPerDay);

    public Clock(int hours, int minutes) => _minutes = (hours* MinutesPerHour + minutes) % MinutesPerDay;

    public Clock Add(int minutesToAdd) => ClockFromMinutes(_minutes + minutesToAdd);

    private Clock ClockFromMinutes(int totalMinutes)
    {
        int hours = totalMinutes / MinutesPerHour;
        int minutes = totalMinutes % MinutesPerHour;
        return new Clock(hours, minutes);
    }

    public Clock Subtract(int minutesToSubtract)
    {
        int totalMinutes = (_minutes - minutesToSubtract) % MinutesPerDay;
        if (totalMinutes < 0)
        {
            totalMinutes = MinutesPerDay + totalMinutes;
        }

        return ClockFromMinutes(totalMinutes);
    }

    public override int GetHashCode() => _minutes.GetHashCode();

    public override string ToString() => $"{_minutes / MinutesPerHour:00}:{_minutes % MinutesPerHour:00}";

    public override bool Equals(object obj) => base.Equals(obj as Clock);

    public bool Equals(Clock other) => other == null ? false : other._minutes == _minutes;
}